### ISSUE TITLE
Хендлер "Регенерация пригласительной ссылки"

### ISSUE THREAD

**Issue by Julia-Ishmuratova** [training-of-new-employees/qon#92]
## Задача

Нужно добавить хендлер для регенерации пригласительной ссылки.

## Описание
_Хендлер:_ `PATCH /api/v1/invitation-link`

Возможность восстановления ссылки доступна только администратору. В запросе передается электронная почта сотрудника (email), который уже зарегистрирован в системе. Кроме того, восстановление выполняется только для сотрудников, которые еще не выполнили первый вход в систему - их учетная запись еще не активирована (active = false).

При регенерации ссылки следует использовать тот же подход, который применяется при генерации ссылки [в функционале "Добавление нового сотрудника"](https://github.com/orgs/training-of-new-employees/projects/1/views/6?pane=issue&itemId=47033589).

Данный хендлер используется в форме "Редактировать данные пользователя" ([макеты](https://www.figma.com/file/xJlE2tj2ErCpKZJGtuEwZv/%D0%9F%D0%BB%D0%B0%D1%82%D1%84%D0%BE%D1%80%D0%BC%D0%B0-%D0%B4%D0%BB%D1%8F-%D0%BE%D0%B1%D1%83%D1%87%D0%B5%D0%BD%D0%B8%D1%8F-%D0%BD%D0%BE%D0%B2%D1%8B%D1%85-%D1%81%D0%BE%D1%82%D1%80%D1%83%D0%B4%D0%BD%D0%B8%D0%BA%D0%BE%D0%B2?type=design&node-id=390-3772&mode=design): Админ. Редактирование пользователя)

### Формат запроса
```
PATCH /api/v1/invitation-link HTTP/1.1
Content-Type: application/json
...

{
  "email": "<email>" // обязательно
}
```
### Возможные коды ответа

- `200` — регенерация ссылки успешно выполнена

_Формат ответа:_
```
{
  "email": "<email>"
  "link": https://{host}/first-login?email={email}&invite={random-string}
}
```
- `400` — неверный формат запроса
- `401` — пользователь неавторизован
- `403` — пользователь не имеет доступа к хендлеру
- `404` — нет пользователя с указанным электронным адресом
- `409` — пользователь уже активирован в системе
- `500` — внутренняя ошибка сервера

_Формат ответа для ошибок:_
```
{
  "error": "<error-message>" // текст ошибки
}
```

**Comment by SurayaBakhishzade** [comment]
Найдены баги: https://app.qase.io/defect/QUICKON/view/85
https://app.qase.io/defect/QUICKON/view/83

**Comment by kozyrev-m** [comment]
https://app.qase.io/defect/QUICKON/view/85

Для регенерации ссылки используется метод `PATCH`

**Comment by kozyrev-m** [comment]
https://app.qase.io/defect/DEMO/view/83

В данном хендлере Ошибка "no access" возвращается только в том случае, если сотрудник имеет отличную компанию от админа.

**Comment by SurayaBakhishzade** [comment]
> https://app.qase.io/defect/QUICKON/view/85
> 
> Для регенерации ссылки используется метод `PATCH`

да, но тут я проверяла, чтобы выходила ошибка 400 так как поменяла метод запроса, но вышла 404

**Comment by kozyrev-m** [comment]
> > https://app.qase.io/defect/QUICKON/view/85
> > Для регенерации ссылки используется метод `PATCH`
> 
> да, но тут я проверяла, чтобы выходила ошибка 400 так как поменяла метод запроса, но вышла 404

У нас хендлер определяется URL-методом и URL-адресом. Согласно указанному методу POST, ручка не найдена (Ошибка 404 указывает на то, что ресурс не найден). Ошибка 400 возвращается, когда хендлер существует, но запрос содержит ошибку.

**Comment by SurayaBakhishzade** [comment]
> > > https://app.qase.io/defect/QUICKON/view/85
> > > Для регенерации ссылки используется метод `PATCH`
> > 
> > 
> > да, но тут я проверяла, чтобы выходила ошибка 400 так как поменяла метод запроса, но вышла 404
> 
> У нас хендлер определяется URL-методом и URL-адресом. Согласно указанному методу POST, ручка не найдена (Ошибка 404 указывает на то, что ресурс не найден). Ошибка 400 возвращается, когда хендлер существует, но запрос содержит ошибку.

А я находила информацию, что 400 может приходить и при неверном URL запроса : изменение метода запроса может вызвать ошибку 400, особенно если API ожидает определенный метод запроса (например, GET, POST, PUT, DELETE) и вы отправляете запрос с неправильным методом.

Например, если API точно указывает, что определенный ресурс поддерживает только GET-запросы, а вы отправляете POST-запрос, сервер может вернуть ошибку 400 с сообщением о том, что метод запроса недопустим.

**Comment by SurayaBakhishzade** [comment]
> https://app.qase.io/defect/QUICKON/view/83
> 
> В данном хендлере ошибка "no access" возвращается только в том случае, если сотрудник имеет отличную компанию от админа.
> 
> Здесь можно только добавить более информативное сообщение. Например, "no access - employee has another company"

Уже добавили это сообщение? Потому что у меня также выходит - no access  @kozyrev-m

**Comment by kozyrev-m** [comment]
> > https://app.qase.io/defect/DEMO/view/83
> > В данном хендлере ошибка "no access" возвращается только в том случае, если сотрудник имеет отличную компанию от админа.
> > Здесь можно только добавить более информативное сообщение. Например, "no access - employee has another company"
> 
> Уже добавили это сообщение? Потому что у меня также выходит - no access [@kozyrev-m](https://github.com/kozyrev-m)

Пока ещё нет. Добавим.

**Comment by kozyrev-m** [comment]
> Здесь можно только добавить более информативное сообщение. Например, "no access - employee has another company"

@SurayaBakhishzade  добавлено
